### PR TITLE
Testing Enumerable Module

### PR DESCRIPTION
**Pull request by phylischumba**
# In this PR we:

- Add specs for all methods created in the Enumerable module.
- Wrapped specs for each method in describe block.
- Used let to define a variable for the test cases.
- Used one it block for each expected behavior.

### CODE REVIEW

File: spec/enumerable_spec.rb [hunk 1]
```diff
@@ -0,0 +1,206 @@
...
+    it 'returns true if an empty array is given' do
+      expect([].my_all?).to be true
+    end
+    it 'returns false if all the items of array does not meet given condition' do
+      expect(num_range.my_all? { |x| x % 4 == 0 }).to be false
+    end
+    it 'returns an error if invoked on object other than enumerable' do
+      expect { 1.my_all? }.to raise_error(NoMethodError)
+    end
+  end
```
Comment (kobiyoyo) [inline spec/enumerable_spec.rb]: Write a test for when a class is passed as an argument

File: spec/enumerable_spec.rb [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+    it 'returns true if an empty array is given' do
+      expect([].my_none?).to be true
+    end
+    it 'returns flase if any of the elements of array meet the condition' do
+      expect(num_arr.my_none? { |x| x < 4 }).to be false
+    end
+    it 'returns an error if invoked on object other than enumerable' do
+      expect { 1.my_none? }.to raise_error(NoMethodError)
+    end
+  end
```
Comment (kobiyoyo) [inline spec/enumerable_spec.rb]: Write a test for when a class is passed as an argument

File: spec/enumerable_spec.rb [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+    it 'returns false if none of elements in a range meet given condition' do
+      expect(num_range.my_any? { |x| x > 9 }).to be false
+    end
+    it 'returns false if an empty array is given' do
+      expect([].my_any?).to be false
+    end
+    it 'returns an error if invoked on object other than enumerable' do
+      expect { 1.my_any? }.to raise_error(NoMethodError)
+    end
+  end
```
Comment (kobiyoyo) [inline spec/enumerable_spec.rb]: Write a test for when a class is passed as an argument

**Review by kobiyoyo — CHANGES_REQUESTED** [review]
## Hi Team     
#### Great Job so far, but there are a few more things required. 
### General
- [x]  Check comments below
### Make changes and submit another code review request.
*Happy coding* 
(Any questions Feel free to contact me on slack @kobiyoyo)

**Review by stratospherique — APPROVED** [review]
### Great work implementing the suggested changes :clap: :clap: Congrats, the project is approved :heavy_check_mark: You may merge :+1: :+1: 

Keep up the good work :muscle: 
**_Ahmed Mahfoudh_**

**Pull request merged**